tools/cldr-code/src/main/java/org/unicode/cldr/util/NameGetter.java

```diff
@@ -0,0 +1,547 @@
+package org.unicode.cldr.util;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.ibm.icu.text.MessageFormat;
+import com.ibm.icu.text.Transform;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NameGetter {
```

Comment:
> isn't it a display name formatter?

Good question! Is that what it is?

> BCP47 does not have @ signs in it, so a different name should be used.

That sounds important. Some of the methods already had comments saying parameters were bcp47 identifiers, and I carried that over to the related methods (callers/callees). If that's inaccurate, what would be a more accurate description? Part of the problem with this whole module is that there are strings and integers coming in without a clear explanation of what strings and integers are actually expected/allowed, and that's why Mark and I have been discussing an Enum instead of strings and integers, wherever that's possible...